### ISSUE TITLE
Template types with default values are not recognized properly.

### ISSUE THREAD

**Issue by stof** [gajus/eslint-plugin-jsdoc#1464]
## Expected behavior

typescript supports defining a default value for a template type in JSDoc (with a syntax inspired from the `@param` default values) since 2021 (implemented in https://github.com/microsoft/TypeScript/pull/45483) but the `jsdoc/valid-types` rule breaks on them.

## Actual behavior

`jsdoc/valid-types` reports an error `Syntax error in namepath: HTMLElement>]`

## ESLint Config

```js
import jsdoc from 'eslint-plugin-jsdoc'

export default [{
    plugins: {
      jsdoc
    },
    settings: {
      jsdoc: {
        mode: 'typescript',
      },
    },
    rules: {
      'jsdoc/valid-types': 'error',
    }
}]
```

## ESLint sample

```js
/**
 * @template {string} Selector
 * @template {keyof GlobalEventHandlersEventMap} TEventType
 * @template {Element} [TElement=import('typed-query-selector/parser').ParseSelector<Selector, HTMLElement>]
 * @param {Selector} selector
 * @param {TEventType} type
 * @param {import('delegate-it').DelegateEventHandler<GlobalEventHandlersEventMap[TEventType], TElement>} callback
 * @param {Omit<AddEventListenerOptions, 'once' | 'signal'>} [options]
 * @returns {void}
 */
export function onGlobalEvent (selector, type, callback, options) {
  delegate(document, selector, type, callback, options)
}
```

## Environment

- Node version: 22.18.0
- ESLint version: 9.33.0
- `eslint-plugin-jsdoc` version: 54.1.1

**Comment by brettz9** [comment]
There is a problem with our parsing which just does a simple split at commas.

For a proper fix, we need something like a `tsdoc` or `typescript-eslint` parser used internally, but the current fix at least checks for whether the whole path is valid. This fix still won't work if you have something complex like `[T=import('abc').Sth<Item1, Item2>, U=number]`, but it at least addresses your use case.
